homeassistant/components/hyperion/config_flow.py

```diff
@@ -0,0 +1,422 @@
+
+    async def async_step_create_token_fail(
+        self, _: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Show an error on the auth form."""
+        # Clean-up the request task.
+        await self._cancel_request_token_task()
+        return self._show_auth_form({CONF_BASE: "auth_new_token_not_granted_error"})
+
+    async def async_step_confirm(
+        self, user_input: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Get final confirmation before entry creation."""
+        if user_input is None and not self._auto_confirm:
+            return await self._show_confirm_form()
+
+        # pylint: disable=no-member  # https://github.com/PyCQA/pylint/issues/3167
+        return self.async_create_entry(
+            title=f"{self._data[CONF_HOST]}:{self._data[CONF_PORT]}", data=self._data
+        )
+
+    def _show_setup_form(self, errors: Optional[Dict] = None) -> Dict[str, Any]:
+        """Show the setup form to the user."""
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST): str,
+                    vol.Optional(CONF_PORT, default=const.DEFAULT_PORT_JSON): int,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    def _show_auth_form(self, errors: Optional[Dict] = None) -> Dict[str, Any]:
+        """Show the auth form to the user."""
+        return self.async_show_form(
+            step_id="auth",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_CREATE_TOKEN): bool,
+                    vol.Optional(CONF_TOKEN): str,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    async def _show_confirm_form(self, errors: Optional[Dict] = None) -> Dict[str, Any]:
+        """Show the confirmation form to the user."""
+        async with await self._create_client() as hyperion_client:
+            if not hyperion_client:
+                return self._show_setup_form({CONF_BASE: "connection_error"})
+            hyperion_id = await hyperion_client.async_id()
+
+        if not hyperion_id:
+            return self.async_abort(reason="no_id")
+
+        await self.async_set_unique_id(hyperion_id)
+        self._abort_if_unique_id_configured()
+
+        if self._auto_confirm:
+            return await self.async_step_confirm()
+
+        return self.async_show_form(
+            step_id="confirm",
+            description_placeholders={
+                CONF_HOST: self._data[CONF_HOST],
+                CONF_PORT: self._data[CONF_PORT],
+                CONF_ID: self.unique_id,
+            },
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the Hyperion Options flow."""
+        return HyperionOptionsFlow(config_entry)
+
+
+class HyperionOptionsFlow(config_entries.OptionsFlow):
```

Comment:
Please keep this PR simpler and keep options flow for a future PR

Reply:
If the options are excluded and saved for a future PR, anyone who migrates from YAML -> ConfigEntry in the intervening time will lose the value of the priority option (which is currently in the YAML).

With that said -- please let me know if you'd rather I proceed with splitting this out and I will do so.

Reply:
We can use a different target branch than the dev branch and do multiple PRs to this other target branch. Then when ready for release we make a PR from the other target branch to dev branch.

Let me know and I can push a copy of dev as a new branch and you can rebase and change target branch here.

Reply:
It could still be imported, just not changed via an options flow.

Reply:
I think the choice is complexity in the codereview (multiple branches) vs complexity in the code (an extra scenario to consider, where the config is migrated but the options not).

Lets go with the new branch if that is okay with you both? @MartinHjelmare